control-flow-functions: Update reference manual links (#16849)
pingcap/docs: functions-and-operators/control-flow-functions.md

# Control Flow Functions

TiDB supports all of the [control flow functions](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html) available in MySQL 8.0.

| Name                                                                                            | Description                       |
|:--------------------------------------------------------------------------------------------------|:----------------------------------|
| [`CASE`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#operator_case)       | Case operator                     |
| [`IF()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_if)         | If/else construct                 |
| [`IFNULL()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_ifnull) | Null if/else construct            |
| [`NULLIF()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_nullif) | Return NULL if expr1 = expr2      |

## CASE

The `CASE` operator allows you to respond on multiple values and conditions.

```sql

> WITH RECURSIVE d AS (SELECT 1 AS n UNION ALL SELECT n+1 FROM d WHERE n<10)
> SELECT n, CASE WHEN n MOD 2 THEN "odd" ELSE "even" END FROM d

Reviewer: ```suggestion
WITH RECURSIVE d AS (SELECT 1 AS n UNION ALL SELECT n+1 FROM d WHERE n<10)
SELECT n, CASE WHEN n MOD 2 THEN "odd" ELSE "even" END FROM d;
```

```sql
WITH data AS (SELECT NULL AS x UNION ALL SELECT 1 )
SELECT x, IFNULL(x,'x has no value') FROM data;
```

```
+------+----------------------------+
| x    | IFNULL(x,'x has no value') |
+------+----------------------------+
| NULL | x has no value             |
|    1 | 1                          |
+------+----------------------------+
2 rows in set (0.0006 sec)
```

## NULLIF()

The `NULLIF()` function returns NULL if both arguments are the same or if the first argument is NULL. Otherwise it returns the first argument.

```sql
WITH RECURSIVE d AS (SELECT 1 AS n UNION ALL SELECT n+1 FROM d WHERE n<10)
SELECT n, NULLIF(n+n, n+2) FROM d;
```

```
+----+------------------+
| n  | NULLIF(n+n, n+2) |
+----+------------------+
|  1 |                2 |
|  2 |             NULL |
|  3 |                6 |
|  4 |                8 |
|  5 |               10 |
|  6 |               12 |
|  7 |               14 |
|  8 |               16 |
|  9 |               18 |
| 10 |               20 |
+----+------------------+
10 rows in set (0.00 sec)
```

> Here n+n and n+2 both evaluate to 4 if n is 2, which makes both arguments the same and the function return NULL.

Reviewer: ```suggestion
In this example, when `n` equals `2`, both `n+n` and `n+2` equal to `4`, making both arguments the same and causing the function to return `NULL`.
```

|  6 | even                                         |
|  7 | odd                                          |
|  8 | even                                         |
|  9 | odd                                          |
| 10 | even                                         |
+----+----------------------------------------------+
10 rows in set (0.00 sec)
```

## IF()

The `IF()` statement allows you to do something based on wheter a value or expression is true or not.

```sql
WITH RECURSIVE d AS (SELECT 1 AS n UNION ALL SELECT n+1 FROM d WHERE n<10)
SELECT n, IF(n MOD 2, "odd", "even") FROM d;
```

```
+----+----------------------------+
| n  | IF(n MOD 2, "odd", "even") |
+----+----------------------------+
|  1 | odd                        |
|  2 | even                       |
|  3 | odd                        |
|  4 | even                       |
|  5 | odd                        |
|  6 | even                       |
|  7 | odd                        |
|  8 | even                       |
|  9 | odd                        |
| 10 | even                       |
+----+----------------------------+
10 rows in set (0.00 sec)
```

## IFNULL()

> The `IFNULL()` functions returns the data of the expression if it is not NULL and if it is NULL it returns the data
> of the second argument of the function.

Author: ```suggestion
The `IFNULL()` functions returns the data of the expression if it is not NULL and if it is NULL it returns the data of the second argument of the function.
```

SELECT n, IF(n MOD 2, "odd", "even") FROM d;
```

```
+----+----------------------------+
| n  | IF(n MOD 2, "odd", "even") |
+----+----------------------------+
|  1 | odd                        |
|  2 | even                       |
|  3 | odd                        |
|  4 | even                       |
|  5 | odd                        |
|  6 | even                       |
|  7 | odd                        |
|  8 | even                       |
|  9 | odd                        |
| 10 | even                       |
+----+----------------------------+
10 rows in set (0.00 sec)
```

## IFNULL()

The `IFNULL()` functions returns the data of the expression if it is not NULL and if it is NULL it returns the data
of the second argument of the function.

```sql
WITH data AS (SELECT NULL AS x UNION ALL SELECT 1 )
SELECT x, IFNULL(x,'x has no value') FROM data;
```

```
+------+----------------------------+
| x    | IFNULL(x,'x has no value') |
+------+----------------------------+
| NULL | x has no value             |
|    1 | 1                          |
+------+----------------------------+
2 rows in set (0.0006 sec)
```

## NULLIF()

> The `NULLIF()` function returns null if both arguments are the same. Otherwise it returns the first argument.

Reviewer 2: Can we clarify what happens when one or more of the arguments are NULL?

Author: ```
mysql> SELECT NULLIF(1, 2);
+--------------+
| NULLIF(1, 2) |
+--------------+
|            1 |
+--------------+
1 row in set (0.00 sec)

mysql> SELECT NULLIF(1, 1);
+--------------+
| NULLIF(1, 1) |
+--------------+
|         NULL |
+--------------+
1 row in set (0.00 sec)

mysql> SELECT NULLIF(1, NULL);
+-----------------+
| NULLIF(1, NULL) |
+-----------------+
|               1 |
+-----------------+
1 row in set (0.00 sec)

mysql> SELECT NULLIF(NULL, NULL);
+--------------------+
| NULLIF(NULL, NULL) |
+--------------------+
|               NULL |
+--------------------+
1 row in set (0.00 sec)

mysql> SELECT NULLIF(NULL, 1);
+-----------------+
| NULLIF(NULL, 1) |
+-----------------+
|            NULL |
+-----------------+
1 row in set (0.00 sec)
```

# Control Flow Functions

TiDB supports all of the [control flow functions](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html) available in MySQL 8.0.

| Name                                                                                            | Description                       |
|:--------------------------------------------------------------------------------------------------|:----------------------------------|
| [`CASE`](#case)       | Case operator                     |
| [`IF()`](#if)         | If/else construct                 |
| [`IFNULL()`](#ifnull) | Null if/else construct            |

> | [`NULLIF()`](#nullif) | Return NULL if expr1 = expr2      |

Reviewer 3: ```suggestion
| [`NULLIF()`](#nullif) | Return `NULL` if expr1 = expr2      |
```

## CASE

> The [`CASE`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#operator_case) operator allows you to perform conditional logic and customize query results based on specified conditions.

Reviewer 3: ```suggestion
The [`CASE`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#operator_case) operator enables you to perform conditional logic and customize query results based on specified conditions.
```

Syntax:

```sql
CASE
    WHEN condition1 THEN result1
    WHEN condition2 THEN result2
    ...
    ELSE default_result
END
```

Example:

```sql
WITH RECURSIVE d AS (SELECT 1 AS n UNION ALL SELECT n+1 FROM d WHERE n<10)
SELECT n, CASE WHEN n MOD 2 THEN "odd" ELSE "even" END FROM d;
```

```
+----+----------------------------------------------+
| n  | CASE WHEN n MOD 2 THEN "odd" ELSE "even" END |
+----+----------------------------------------------+
|  1 | odd                                          |
|  2 | even                                         |
|  3 | odd                                          |
|  4 | even                                         |
|  5 | odd                                          |
|  6 | even                                         |
|  7 | odd                                          |
|  8 | even                                         |
|  9 | odd                                          |
| 10 | even                                         |
+----+----------------------------------------------+
10 rows in set (0.00 sec)
```

## IF()

> The [`IF()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_if) function allows you to perform different actions based on whether a value or expression is true or not.

Reviewer 3: ```suggestion
The [`IF()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_if) function enables you to perform different actions based on whether a value or expression is true or not.
```

# Control Flow Functions

TiDB supports all of the [control flow functions](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html) available in MySQL 8.0.

| Name                                                                                            | Description                       |
|:--------------------------------------------------------------------------------------------------|:----------------------------------|
| [`CASE`](#case)       | Case operator                     |

> | [`IF()`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_if)         | If/else construct                 |

Reviewer: ```suggestion
| [`IF()`](#if)         | If/else construct                 |
```

> | [`IFNULL()`](#if) | Null if/else construct            |

Reviewer: ```suggestion
| [`IFNULL()`](#ifnull) | Null if/else construct            |
```

Example:

```sql
WITH data AS (SELECT NULL AS x UNION ALL SELECT 1 )
SELECT x, IFNULL(x,'x has no value') FROM data;
```

```
+------+----------------------------+
| x    | IFNULL(x,'x has no value') |
+------+----------------------------+
| NULL | x has no value             |
|    1 | 1                          |
+------+----------------------------+
2 rows in set (0.0006 sec)
```

## NULLIF()

The [`NULLIF(expr1,expr2)`](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_nullif) function returns `NULL` if both arguments are the same or if the first argument is `NULL`. Otherwise, it returns the first argument.

Example:

```sql
WITH RECURSIVE d AS (SELECT 1 AS n UNION ALL SELECT n+1 FROM d WHERE n<10)
SELECT n, NULLIF(n+n, n+2) FROM d;
```

```
+----+------------------+
| n  | NULLIF(n+n, n+2) |
+----+------------------+
|  1 |                2 |
|  2 |             NULL |
|  3 |                6 |
|  4 |                8 |
|  5 |               10 |
|  6 |               12 |
|  7 |               14 |
|  8 |               16 |
|  9 |               18 |
| 10 |               20 |
+----+------------------+
10 rows in set (0.00 sec)
```

> In this example, when `n` equals `2`, both `n+n` and `n+2` equal to `4`, making both arguments the same and causing the function to return `NULL`.

Reviewer: ```suggestion
In this example, when `n` equals `2`, both `n+n` and `n+2` equal `4`, making both arguments the same and causing the function to return `NULL`.
```
